File: December/19th December,2024/Leetcode(769).js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maxChunksToSorted = function(arr) {
    let ans=0;
    let m=0;
    for(let i=0;i<arr.length;i++){
        m=Math.max(arr[i],m);
        if(m<=i) ans++;
    }
    return ans;
};

File: December/19th December,2024/Leetcode(769).java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int ans = 0, m = 0;
        for (int i = 0; i < arr.length; i++) {
            m = Math.max(arr[i], m);
            if (m <= i)
                ans++;
        }
        return ans;
    }
}